Tigger/Coordinates.py

```diff
@@ -341,28 +341,110 @@ class FITSWCS(FITSWCSpix):
         def __init__(self, header):
             """Constructor. Create from filename (treated as FITS file), or a FITS header object"""
             Projection.FITSWCSpix.__init__(self, header)
+            # init() has been modified to be a self contained workaround for Tigger v1.6.0
+            # get number of axis
+            naxis = header['NAXIS']
+
+            # get refpix
+            self.refpix = []
+            for iaxis in range(naxis):
+                if iaxis < 2:
+                    self.refpix.append(header["CRPIX%d" % (iaxis + 1)] - 1)
+                else:
+                    self.refpix.append(header["CRPIX%d" % (iaxis + 1)])
+
+            # get ra and dec axis
+            self.ra_axis = self.dec_axis = None
+            for iaxis in range(naxis):
+                name = header.get("CTYPE%d" % (iaxis + 1), '').upper()
+                if name.startswith("RA"):
+                    self.ra_axis = iaxis
+                elif name.startswith("DEC"):
+                    self.dec_axis = iaxis
+
+            # use WCS pixel information to build refsky
+            self.wcs = WCS(header)
+            wcs_pixel = self.wcs.pixel_shape
+            refsky_pixels = []
+            for ipixel in range(naxis):
+                if ipixel < 2:
+                    refsky_pixels.append(int(wcs_pixel[ipixel] / 2))
+                else:
+                    refsky_pixels.append(wcs_pixel[ipixel])
+            self.refsky = self.wcs.wcs_pix2world([refsky_pixels], 0)[0, :]
+
+            # set centre x/y pixels
+            self.xpix0, self.ypix0 = self.refpix[self.ra_axis], self.refpix[self.dec_axis]
+
+            # set x/y scales
+            pixscale = (header["CDELT1"], header["CDELT2"])
+            self.xscale = pixscale[self.ra_axis] * DEG
+            self.yscale = pixscale[self.dec_axis] * DEG
+
+            has_projection = True
             self._l0 = self.refpix[self.ra_axis]
             self._m0 = self.refpix[self.dec_axis]
 
         def lm(self, ra, dec):
-            l, m = Projection.FITSWCSpix.lm(self, ra, dec)
-            return sin((l - self._l0) * self.xscale), sin((m - self._m0)*self.yscale)
```

Comment:
I strongly believe this and the alternative implementation is very wrong. Following Eric Greisen's derivation of the various coordinate systems for AIPS. Memo 27 Eqns 10

ftp://ftp.aoc.nrao.edu/pub/software/aips/TEXT/PUBL/AIPSMEMO27.PS

Reply:
@o-smirnov can you please comment on this